spec/Checker/Eligibility/ShippingSubscriptionEligibilityCheckerSpec.php

```diff
@@ -0,0 +1,138 @@
+    }
+
+    function it_should_not_display_shipping_method_without_subscription(
+        ShipmentInterface $shippingSubject,
+        ShippingSubscriptionMethodInterface $shippingMethod,
+        OrderInterface $order,
+        SubscriptionAwareInterface $subscriptionAware,
+        SubscriptionExpirationCheckerInterface $subscriptionExpirationChecker
+    )
+    {
+        $order->getCustomer()->willReturn($subscriptionAware);
+        $order->getTotal()->willReturn(20000);
+
+
+        $shippingSubject->getOrder()->willReturn($order);
+
+        $subscriptionExpirationChecker->checkSubscription($subscriptionAware)->willReturn(false);
+
+        $shippingMethod->getAvailableFromTotal()->willReturn(20000);
+        $shippingMethod->isShippingSubscription()->willReturn(true);
+
+        $this->isSubscriptionActive($shippingSubject, $shippingMethod)->shouldReturn(false);
+
+    }
+
+    function it_should_not_display_shipping_method_without_minimum_order_total(
+        ShipmentInterface $shippingSubject,
+        ShippingSubscriptionMethodInterface $shippingMethod,
+        OrderInterface $order,
+        SubscriptionAwareInterface $subscriptionAware,
+        SubscriptionExpirationCheckerInterface $subscriptionExpirationChecker
+    )
+    {
+        $order->getCustomer()->willReturn($subscriptionAware);
+        $order->getTotal()->willReturn(19999);
+
+
+        $shippingSubject->getOrder()->willReturn($order);
+
+        $subscriptionExpirationChecker->checkSubscription($subscriptionAware)->willReturn(true);
+
+        $shippingMethod->getAvailableFromTotal()->willReturn(20000);
+        $shippingMethod->isShippingSubscription()->willReturn(true);
+
+        $this->isSubscriptionActive($shippingSubject, $shippingMethod)->shouldReturn(false);
+
+    }
+
+    function it_should_display_shipping_method_if_subscription_is_not_required(
+        ShipmentInterface $shippingSubject,
+        ShippingSubscriptionMethodInterface $shippingMethod,
+        OrderInterface $order,
+        SubscriptionAwareInterface $subscriptionAware,
+        SubscriptionExpirationCheckerInterface $subscriptionExpirationChecker
+    )
+    {
+        $order->getCustomer()->willReturn($subscriptionAware);
+        $order->getTotal()->willReturn(19999);
+
+        $shippingSubject->getOrder()->willReturn($order);
+
+        $subscriptionExpirationChecker->checkSubscription($subscriptionAware)->willReturn(false);
+
+        $shippingMethod->getAvailableFromTotal()->willReturn(20000);
+        $shippingMethod->isShippingSubscription()->willReturn(false);
+
+        $this->isSubscriptionActive($shippingSubject, $shippingMethod)->shouldReturn(true);
+    }
+
+    function it_should_display_shipping_method_if_customer_has_subscription_and_minimum_order_total(
+        ShipmentInterface $shippingSubject,
+        ShippingSubscriptionMethodInterface $shippingMethod,
+        OrderInterface $order,
+        SubscriptionAwareInterface $subscriptionAware,
```

Comment:
We need better name for $subscriptionAware maybe better would be $customer

Reply:
Changed from $subscriptionAware to $customer